src/common/buf.c

```diff
@@ -5,6 +5,15 @@
 #include "common/buf.h"
 #include "common/mem.h"
 
+static void
+buf_replace_by(struct buf *dst, struct buf *src)
+{
+	free(dst->buf);
+	dst->buf = src->buf;
```

Comment:
Maybe reset `src->buf` to `NULL` also?

Reply:
Replaced everything in the src buffer by `*src = (struct buf){0}`.

Also added some asserts which try to ensure that the dst buffer is either cleared out or initialized, otherwise supplying just a stack allocated dst buffer would cause us to try to free a random memory address.

Reply:
Trying to detect an uninitialized dst buffer seems like it may not be reliable (the garbage values might look valid). So I would lean toward omitting those checks. OK with me either way though.

Reply:
> Trying to detect an uninitialized dst buffer seems like it may not be reliable (the garbage values might look valid).

Agree, we'd need to have some defined magic number in one of the fields and at that point we are way overkill for a static function that is only called within the same small `.c` file. I'll remove them.